packages/app/src/test-utils.tsx

```diff
@@ -1,28 +1,47 @@
-import { act, render, screen, within } from '@testing-library/react';
+import type { RenderResult } from '@testing-library/react';
+import { render, screen, within } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
+import type { UserEvent } from '@testing-library/user-event/dist/types/setup';
 
 import App from './App';
 import MockProvider from './providers/mock/MockProvider';
 import type { Vis } from './vis-packs/core/visualizations';
 
-export async function renderApp(): Promise<void> {
-  return act(async () => {
-    render(
-      <MockProvider>
-        <App />
-      </MockProvider>
-    );
-  });
+interface RenderAppResult extends RenderResult {
+  user: UserEvent;
+  selectExplorerNode: (path: string) => Promise<void>;
+  selectVisTab: (name: Vis) => Promise<void>;
 }
 
-export async function selectExplorerNode(path: string): Promise<void> {
-  for await (const segment of path.split('/')) {
-    userEvent.click(
-      await screen.findByRole('treeitem', {
-        name: new RegExp(`^${segment}(?: \\(NeXus group\\))?$`, 'u'), // account for potential NeXus badge
-      })
-    );
-  }
+export async function renderApp(): Promise<RenderAppResult> {
+  const user = userEvent.setup({ delay: null }); // https://github.com/testing-library/user-event/issues/833
+  const renderResult = render(
+    <MockProvider>
+      <App />
+    </MockProvider>
+  );
+
+  await screen.findByLabelText('Loading root metadata...');
+  await screen.findByRole('treeitem', { name: 'entities' });
```

Comment:
The weird async workaround we had in `renderApp`: `act(async () => { render(...) })`, was no longer working. I managed to fix the tests with the above, which is no longer a workaround, really -- we just wait for the initial metadata to finish loading, which makes sense.